app/src/main/kotlin/com/flxrs/dankchat/data/twitch/message/PrivMessage.kt

```diff
@@ -81,4 +84,16 @@ val PrivMessage.isFirstMessage: Boolean
     get() = tags["first-msg"] == "1"
 
 val PrivMessage.isElevatedMessage: Boolean
-    get() = tags["pinned-chat-paid-amount"] != null
\ No newline at end of file
+    get() = tags["pinned-chat-paid-amount"] != null
+
+/** format name for display in chat */
+val PrivMessage.fullName: String
+    get() = userDisplay?.alias ?: when {
+        displayName.equals(name, ignoreCase = true) -> displayName
+        else                                        -> "$name($displayName)"
+    }
+
+fun PrivMessage.finalColor(@ColorInt bgColor: Int): Int {
```

Comment:
The naming here doesn't really tell what this function does, e.g. the fallback and color normalization.
Perhaps:
```suggestion
fun PrivMessage.customOrUserColor(@ColorInt background: Int): Int {
```

Reply:
I go with `customOrUserColorOn` to indicate more clearly than the passed color is background, not some kind of fallback value like "getOrElse" kind of functions.